File: solutions/cpp/lasagna-master/1/lasagna_master.cpp
#include "lasagna_master.h"
 
namespace lasagna_master {
    int preparationTime(const std::vector<std::string>& layers, int averageTime) {
        return averageTime * layers.size();
    }

    amount quantities(const std::vector<std::string>& layers) {
        amount needed(0, 0);
        for (std::string layer: layers) {
            if (layer == "sauce")
                needed.sauce += 0.2;

            if (layer == "noodles")
                needed.noodles += 50;
        }
        return needed;
    }

    void addSecretIngredient(std::vector<std::string>& myList, const std::vector<std::string>& friendsList) {
        myList.back() = friendsList.back();
    }

    std::vector<double> scaleRecipe(const std::vector<double>& quantities, int scale) {
        std::vector<double> scaled;
        scaled.reserve(quantities.size());
        for(double quantity: quantities) {
            scaled.push_back(scale * quantity / 2.);
        }
        return scaled;
    }

    void addSecretIngredient(std::vector<std::string>& myList, std::string secretIngredient) {
        myList.back() = secretIngredient;
    }

}  // namespace lasagna_master

File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, preparationTime int) int {
    if preparationTime == 0 {
        preparationTime = 2;
    }
	return len(layers) * preparationTime
}

// TODO: define the 'Quantities()' function
func Quantities(layers []string) (int, float64) {
    noodles, sauce := 0, 0.;
    for i := 0; i < len(layers); i++ {
        switch layers[i] {
            case "noodles":
        		noodles++;
            	break;
            case "sauce":
        		sauce++;
            	break;
        }
    }
    return 50 * noodles, 0.2 * sauce
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendsList, myList []string) []string {
    myList[len(myList) - 1] = friendsList[len(friendsList) - 1];
    return myList
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(quantities []float64, portions int) []float64 {
    scaled := make([]float64, len(quantities));
    for i := 0; i < len(quantities); i++ {
        scaled[i] = quantities[i] * float64(portions) / 2;
    }
	return scaled
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
// 
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more 
// functionality.
